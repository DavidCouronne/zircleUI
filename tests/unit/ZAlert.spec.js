import { shallowMount, createLocalVue } from '@vue/test-utils'
import zalert from '@/components/z-alert.vue'
import zslider from '@/components/z-slider.vue'
import zircle from '@/index'
const localVue = createLocalVue()
localVue.use(zircle)
const wrapper = shallowMount(zalert, {
  propsData: {
    slider: true,
    color: 'success'
  },
  slots: {
    default: '<p> lorem text </p>',
    extension: '<div></div>'
  },
  stubs: {
    'z-slider': zslider
  },
  localVue
})
describe('z-alert.vue', () => {
  it('Renders props when passed', () => {
    expect(wrapper.vm.color).toEqual('success')
  })

  it('Has z-slider component', () => {
    expect(wrapper.find(zslider).exists()).toBe(true)
  })
  it('Has the expected html structure', () => {
    // Note: If props.imagesrc and or props.label are present, slots.imagesrc and slots.label wont be rendered
    expect(wrapper.element).toMatchSnapshot()
  })
})
