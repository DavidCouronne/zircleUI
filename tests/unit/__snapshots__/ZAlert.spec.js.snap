// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`z-alert.vue Has the expected html structure 1`] = `
<div
  class="zui pop success"
  name="z-alert"
  style="width: 250px; height: 250px; margin: -125px 0px 0px -125px;"
  type="alert"
>
  <svg
    class="slider"
    viewBox="0 0 100 100"
    xmlns="http://www.w3.org/2000/svg"
  >
    <circle
      cx="50"
      cy="50"
      r="51"
      style="transform: rotate(-90deg); stroke-dasharray: 314.1592653589793; stroke-dashoffset: 314.1592653589793;"
    />
  </svg>
   
  <div
    class="z-popup-plate"
    style="width: 380px; height: 380px; margin: -190px 0px 0px -190px;"
  />
   
  <div
    class="z-content"
  >
    <p>
       lorem text 
    </p>
  </div>
   
  <div />
</div>
`;
